### PR TITLE
Added 2 tests aimed at verifying '_fields' option and its parameters …

### PR DESCRIPTION
**Pull request by andriikonstantinov**
…from server's and client's side

### CODE REVIEW

**Comment by marcin-ptaszynski** [comment]
@andriikonstantinov , too long commit message. Please keep first line within 50 characters.

**Comment by marcin-ptaszynski** [comment]
@andriikonstantinov , GET singular doesn't apply the `_fields` parameter.

File: db/transaction/transaction.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ type Transaction interface {
 	Update(*schema.Resource) error
 	StateUpdate(*schema.Resource, *ResourceState) error
 	Delete(*schema.Schema, interface{}) error
-	Fetch(*schema.Schema, Filter) (*schema.Resource, error)
+	Fetch(*schema.Schema, Filter, *ListOptions) (*schema.Resource, error)
```
Comment (marcin-ptaszynski) [inline db/transaction/transaction.go]: They're not *List* Options anymore. Perhaps `ViewOptions` or `DetailsOptions`?

Comment (morrisson) [inline db/transaction/transaction.go]: Should add same options to LockFetch

File: server/resources/resource_management.go [hunk 2]
```diff
@@ -313,6 +309,11 @@ func GetSingleResource(context middleware.Context, dataStore db.DB, resourceSche
 	if err != nil {
 		return err
 	}
+	var options *transaction.ListOptions
```
Comment (morrisson) [inline server/resources/resource_management.go]: Why is this logic in "GetSingleResource"? In the List function, this logic is in "GetResourcesInTransaction", so this logic should be in "GetSingleResourceInTransaction" unless there is a special reason.

File: db/db.go [hunk 3]
```diff
@@ -90,7 +90,7 @@ func CopyDBResources(input, output DB, overrideExisting bool) error {
 
 		for _, resource := range resources {
 			log.Info("Creating resource %s", resource.ID())
-			destResource, _ := otx.Fetch(s, transaction.IDFilter(resource.ID()))
+			destResource, _ := otx.Fetch(s, transaction.IDFilter(resource.ID()),nil)
```
Comment (morrisson) [inline db/db.go]: should add whitespace after comma.

**Comment by przemyslaw-dobrowolski-cl** [comment]
@andriikonstantinov please rebase

File: db/transaction/transaction.go [hunk 4]
```diff
@@ -59,8 +59,8 @@ type ResourceState struct {
 	Monitoring    string
 }
 
-//ListOptions specifies additional list related options.
-type ListOptions struct {
+//ViewOptions specifies additional list related options.
```
Comment (p-kozlowski) [inline db/transaction/transaction.go]: it's not "list" anymore (same elsewhere). this is why I dislike comments in the code.

File: extension/otto/gohan_db_test.go [hunk 5]
```diff
@@ -284,7 +285,7 @@ var _ = Describe("GohanDb", func() {
 					)
 				},
 				Entry("gohan_db_list", "gohan_db_list", "List"),
-				Entry("gohan_db_lock_list", "gohan_db_lock_list", "LockList"),
+				//Entry("gohan_db_lock_list", "gohan_db_lock_list", "LockList"),
```
Comment (p-kozlowski) [inline extension/otto/gohan_db_test.go]: test case commented out

File: server/resources/resource_management.go [hunk 6]
```diff
@@ -385,8 +386,7 @@ func GetSingleResourceInTransaction(context middleware.Context, resourceSchema *
 	if tenantIDs != nil {
 		filter["tenant_id"] = tenantIDs
 	}
-	object, err := mainTransaction.Fetch(resourceSchema, filter)
-
+	object, err := mainTransaction.Fetch(resourceSchema, filter,options)
```
Comment (p-kozlowski) [inline server/resources/resource_management.go]: missing space (here and elsewhere). go fmt

Comment (andriikonstantinov) [inline server/resources/resource_management.go]: Sure.Done

File: db/transaction/transaction.go (hunk 4 above)
Comment (przemyslaw-dobrowolski-cl) [inline db/transaction/transaction.go]: BTW, how do you refactor naming? by hand? InteliJ has a nice and working rename tool which also renames usages in comments.

Comment (andriikonstantinov) [inline db/transaction/transaction.go]: No.I used this tool.I think @p-kozlowski meant that "ViewOptions specifies additional list related options." didn't make sense anymore,that's why I changed this sentence by hand.

File: server/resources/resource_management.go (hunk 2 above)
Comment (andriikonstantinov) [inline server/resources/resource_management.go]: @morrisson Changed

**Comment by przemyslaw-dobrowolski-cl** [comment]
@andriikonstantinov please rebase

**Comment by p-kozlowski** [comment]
@morrisson, @nati please review
@andriikonstantinov please rebase

**Comment by andriikonstantinov** [comment]
@morrisson please review

File: db/db.go (hunk 3 above)
Comment (andriikonstantinov) [inline db/db.go]: Done

**Pull request merged**